src/sbd-cluster.c

```diff
@@ -67,21 +70,149 @@ sbd_plugin_membership_dispatch(cpg_handle_t handle,
 #endif
 
 #if SUPPORT_COROSYNC
+#include <corosync/cmap.h>
+
+static bool two_node = false;
+static bool ever_seen_both = false;
+static cmap_handle_t cmap_handle = 0;
+static cmap_track_handle_t track_handle = 0;
+static int cpg_membership_entries = -1;
+static GSource *cmap_source = NULL;
+
+void
+sbd_cpg_membership_health_update()
+{
+    if(cpg_membership_entries > 0) {
+        if (!two_node || !ever_seen_both || cpg_membership_entries > 1) {
```

Comment:
Something seems off here.

Either `ever_seen_both` is badly named ('both' implies 2 nodes) or perhaps `!two_node` should be just `two_node` here.  Or possibly both, I'm really not sure what you're going for here.  

It seems like a 5 node cluster with 1 active node would end up setting `pcmk_health_online`, even if it had previously been able to see everyone.  Which seems wrong.  Maybe you want s/||/&&/ ?

Might be a good idea to include a comment describing the scenario you're trying to detect.